crates/hotshot/src/traits/election/randomized_committee.rs

```diff
@@ -0,0 +1,166 @@
+                .collect();
+
+        // For each member, get the stake table entry
+        let members: Vec<<TYPES::SignatureKey as SignatureKey>::StakeTableEntry> =
+            committee_members
+                .iter()
+                .map(|member| member.stake_table_entry.clone())
+                .filter(|entry| entry.stake() > U256::zero())
+                .collect();
+
+        // Index the stake table by public key
+        let indexed_stake_table: BTreeMap<
+            TYPES::SignatureKey,
+            <TYPES::SignatureKey as SignatureKey>::StakeTableEntry,
+        > = members
+            .iter()
+            .map(|entry| (TYPES::SignatureKey::public_key(entry), entry.clone()))
+            .collect();
+
+        Self {
+            eligible_leaders,
+            stake_table: members,
+            indexed_stake_table,
+            committee_topic,
+        }
+    }
+
+    /// Get the stake table for the current view
+    fn stake_table(
+        &self,
+    ) -> Vec<<<TYPES as NodeType>::SignatureKey as SignatureKey>::StakeTableEntry> {
+        self.stake_table.clone()
+    }
+
+    /// Get all members of the committee for the current view
+    fn committee_members(
+        &self,
+        _view_number: <TYPES as NodeType>::Time,
+    ) -> std::collections::BTreeSet<<TYPES as NodeType>::SignatureKey> {
+        self.stake_table
+            .iter()
+            .map(TYPES::SignatureKey::public_key)
+            .collect()
+    }
+
+    /// Get all eligible leaders of the committee for the current view
+    fn committee_leaders(
+        &self,
+        _view_number: <TYPES as NodeType>::Time,
+    ) -> std::collections::BTreeSet<<TYPES as NodeType>::SignatureKey> {
+        self.eligible_leaders
+            .iter()
+            .map(TYPES::SignatureKey::public_key)
+            .collect()
+    }
+
+    /// Get the stake table entry for a public key
+    fn stake(
+        &self,
+        pub_key: &<TYPES as NodeType>::SignatureKey,
+    ) -> Option<<TYPES::SignatureKey as SignatureKey>::StakeTableEntry> {
+        // Only return the stake if it is above zero
+        self.indexed_stake_table.get(pub_key).cloned()
+    }
+
+    /// Check if a node has stake in the committee
+    fn has_stake(&self, pub_key: &<TYPES as NodeType>::SignatureKey) -> bool {
+        self.indexed_stake_table
+            .get(pub_key)
+            .is_some_and(|x| x.stake() > U256::zero())
+    }
+
+    /// Get the network topic for the committee
+    fn committee_topic(&self) -> Topic {
+        self.committee_topic.clone()
+    }
+
+    /// Index the vector of public keys with the current view number
+    fn leader(&self, view_number: TYPES::Time) -> TYPES::SignatureKey {
+        let mut rng: StdRng = rand::SeedableRng::seed_from_u64(*view_number);
+        let randomized_view_number: usize = rng.gen();
+        let index = randomized_view_number % self.eligible_leaders.len();
+        let res = self.eligible_leaders[index].clone();
```

Comment:
Question for my understanding: Is there a downside to doing `rng.gen_range` here?

Reply:
Oh good catch.

It looks like `gen` doesn't have any guarantees about the distribution it uses, so I think `gen_range` is the correct thing to use. updated